src/py42/_internal/client_factories.py

```diff
@@ -111,6 +115,15 @@ def get_alert_rules_client(self):
             )
         return self._alert_rules_client
 
+    def get_saved_search_client(self):
+        if not self._saved_search_client:
+            if not self._file_event_session:
```

Comment:
Opportunity for making a `_get_file_event_session()` function that handles the lazy setting / returning and can be used by `get_file_event_client()` and `get_saved_search_client()`